### PR TITLE
Add more setup instructions to `@metamask/typescript-eslint-config` README

### PR DESCRIPTION
**Pull request by Gudahtt**
Additional instructions have been added to explain how to configure ESLint to support rules that require type information.

### CODE REVIEW

**Comment by Gudahtt** [comment]
If we want to remove the rule instead, we can close this PR.

**Comment by Gudahtt** [comment]
Moving this back into draft because we're going to disable the rule for now, to make updating to v9 easier. Type information ESLint rules will come in a future release, at which point I'll bring this back out of draft.

**Comment by Gudahtt** [comment]
Related to https://github.com/MetaMask/eslint-config/pull/210

**Comment by Gudahtt** [comment]
We have delayed rules requiring type information until v11. This can be revisited once v10 has been published.

**Comment by Gudahtt** [comment]
@Mrtenz I just noticed this PR was still in draft. Does this look correct to you? I haven't tried these instructions out recently.

File: packages/typescript/README.md [hunk 1]
```diff
@@ -39,6 +39,22 @@ module.exports = {
       files: ['*.ts'],
       extends: ['@metamask/eslint-config-typescript'],
     },
+    // An alternative `tsconfig` file for tests is specified here so that test files can be linted
+    // without building them during the normal build process. This file should have `noEmit` set to
+    // `true`.
+    {
+      files: ['*.test.ts'],
+      parserOptions: {
+        project: './tsconfig.test.json',
```
Comment (Gudahtt) [inline packages/typescript/README.md:48]: Not sure we need this. Our conventions have been drifting to using `tsconfig.json` as a `noEmit` config just for testing, and using `tsconfig.build.json` for building. In which case this section is not necessary.

File: packages/typescript/README.md [hunk 2]
```diff
@@ -39,6 +39,22 @@ module.exports = {
...
+      },
+    },
```
Comment (Gudahtt) [inline packages/typescript/README.md:50]: Not sure we need this. Our conventions have been drifting to using `tsconfig.json` as a `noEmit` config just for testing, and using `tsconfig.build.json` for building. In which case this section is not necessary.

```suggestion
```

File: packages/typescript/README.md [hunk 3]
```diff
@@ -39,6 +39,22 @@ module.exports = {
...
   ],
+
+  // This is required for rules that use type information.
+  // See here for more information: https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md
+  parserOptions: {
+    project: './tsconfig.json',
```
Comment (Mrtenz) [inline packages/typescript/README.md:56]: This doesn't need to be set, unless a different `tsconfig.json` is required. We default to `tsconfig.json`.

**Review by Mrtenz — CHANGES_REQUESTED** [review]
Just a small comment.

File: packages/typescript/README.md (hunk 3 above)
Comment (Gudahtt) [reply packages/typescript/README.md:47]: Good point!
```suggestion
```

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**